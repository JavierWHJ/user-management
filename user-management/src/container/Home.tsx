import React, {useState, useEffect} from 'react';
import userService from "../services/userService";
import { User } from '../type';
import UserContainer from './UserContainer';

const Home = () => {
    const [users, setUsers] = useState<User[]>();

    useEffect(() => {
        getUsers();
    }, [])

    const getUsers = () => {
        userService.fetchUsers().then((users) => {
            setUsers(users)
        })
    }

    const addUser = (newUser: User) => {
        userService.addUser(newUser).then((users) => {
            getUsers();
        })
    }

    return (
        <div>
            <UserContainer users={users}/>
        </div>
    );
}

export default Home;
