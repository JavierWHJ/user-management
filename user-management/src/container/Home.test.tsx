import React from "react";
import Home from "./Home";

import { render } from "@testing-library/react";

it("matches snapshot", () => {
  const { asFragment } = render(<Home />);

  expect(asFragment()).toMatchSnapshot();
});