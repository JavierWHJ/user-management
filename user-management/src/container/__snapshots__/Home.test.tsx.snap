// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="mt-4 d-flex justify-content-between"
      >
        <h2>
          Users
        </h2>
        <button
          class="btn btn-primary"
          type="button"
        >
          Add User
        </button>
      </div>
      <div>
        <input
          class="form-control form-control-lg mt-3"
          name="searchterm"
          placeholder="Search User"
          type="text"
        />
      </div>
      <div
        class="fade alert alert-info show"
        role="alert"
        style="margin-top: 20px;"
      >
        There are no users.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
